roleoption: remove sqltelemetry dependency

otan opened the pull request:
For GetStmts, instead use a function hook to inject any necessary
telemetry calls. This is one more step towards removing `util/log`
dependencies from the `tree` package.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/80895)
<!-- Reviewable:end -->

ajwerner reviewed (commented):
You've got a test that's sad about a missing `strings.ToLower`, otherwise <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/80895#-:-N16cr2_2gfNqsfrfaYE:bq4pxfe)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @RichardJCai)



<!-- Sent from Reviewable.io -->

otan committed:
roleoption: remove sqltelemetry dependency

For GetStmts, instead use a function hook to inject any necessary
telemetry calls. This is one more step towards removing `util/log`
dependencies from the `tree` package.

Release note: None

otan commented:
woops, got it.

bors r=ajwerner

otan commented:
bors r=ajwerner

flake with [TestReplicaTombstone](https://teamcity.cockroachdb.com/viewLog.html?buildId=5082248&buildTypeId=Cockroach_UnitTests#testNameId-83984481530307854)
